### DISCUSSION TITLE
Thank you very much for using CMAKE!

### DISCUSSION THREAD

**Discussion by dougdew64** [ml-explore/mlx#86] [General]
Whomever decided to use CMAKE for MLX is my favorite person today, along with the folks who created CMAKE.

It is super easy to perform this sequence of 10 steps to begin a weekend of enjoying MLX code in the Xcode debugger:

1. Create an "Xcode" subdirectory under the MLX git root directory
2. CD into the Xcode subdirectory and run cmake -G Xcode .. to generate an mlx.xcodeproj file
3. Create an Xcode workspace
4. Add the mlx.xcodeproj file to the Xcode workspace
5. Add a new (second) C++ command project to the Xcode workspace
6. Configure the C++ command project to point to the header directories, library directories, add the libmlx.a library and set a preprocessor macro to point to  the mlx.metallib file
7. Add the tutorial.cpp file to the C++ command project
8. Build
9. Set a breakpoint in tutorial.cpp
10. Start the weekend!

Thank you again MLX team!

![image](https://github.com/ml-explore/mlx/assets/98898390/6ef986c6-caa3-4e82-9cfb-adce3112b0d4)

**Comment by abksdev** [comment]
Hi @dougdew64,

Do you know how to link a pre-build libmlx.a from a `CMAKE_INSTALL_PREFIX=/Users/home_dir/Developer/mlX/mlX_cpp_libs`

I am using below cmd and getting `ld: Undefined symbols:` Error.
```
/usr/bin/g++  tutorial.cpp -o tutorial -I/Users/home_dir/Developer/mlX/mlX_cpp_libs/include --std=c++17 -DMETAL_PATH=/Users/home_dir/Developer/mlX/mlX_cpp_libs/lib -L/Users/home_dir/Developer/mlX/mlX_cpp_libs/lib -lmlx
```
can you please help me with this or how do the same via Xcode.
</br>
Thanks in advance

**Comment by dougdew64** [comment]
@abksdev Here is my Xcode configuration.

In my Xcode workspace, my "tutorial" project is the MLX c++ tutorial project. I've linked my "tutorial" project to the CMAKE build of "mlx."

![image](https://github.com/ml-explore/mlx/assets/98898390/06d87953-affe-4ff0-9963-457b96404483)
![image](https://github.com/ml-explore/mlx/assets/98898390/b28e81e1-1c6b-4aee-b3fa-ef77e7ac71ad)
